pkg/source/adapter/adapter.go

```diff
@@ -136,6 +136,7 @@ func (a *Adapter) Handle(ctx context.Context, msg *sarama.ConsumerMessage) (bool
 		a.logger.Debug("Error while sending the message", zap.Error(err))
 		return false, err // Error while sending, don't commit offset
 	}
+	res.Body.Close()
```

Comment:
how about `nil` check ? 

and move that to a function, eg:

```
func safeCloseHTTPResponseBody(response *http.Response) 
  if response != nil && response.Body != nil {
    err := response.Body.Close()
    if err != nil {
      c.logger.Error("Failed To Close HTTP Response Body", zap.Error(err))
    }
  }
}
```

than `	defer safeCloseHTTPResponseBody(response)`  instead of the direct ? 


@slinkydeveloper have we smth like that already in the kncloudevents ? :thinking:

Reply:
Yep you miss a nil check, but i don't think we need to do a check on the body.Close error. If there's a broken pipe we don't care anyway, we don't read that body.

> @slinkydeveloper have we smth like that already in the kncloudevents ?

We don't have such function